dev/tools/localization/localizations_utils.dart

```diff
@@ -371,46 +371,46 @@ String generateClassDeclaration(
 class $classNamePrefix$camelCaseName extends $superClass {''';
 }
 
-/// Return `s` as a Dart-parseable raw string in single or double quotes.
+/// Return `s` as a Dart-parseable string.
 ///
-/// Double quotes are expanded:
+/// The result tries to avoid character escaping:
 ///
 /// ```
-/// foo => r'foo'
-/// foo "bar" => r'foo "bar"'
-/// foo 'bar' => r'foo ' "'" r'bar' "'"
+/// foo => 'foo'
+/// foo "bar" => 'foo "bar"'
+/// foo 'bar' => "foo 'bar'"
+/// foo 'bar' "baz" => '''foo 'bar' "baz"'''
+/// foo\bar => r'foo\bar'
 /// ```
-String generateString(String s) {
-  if (!s.contains("'"))
-    return "r'$s'";
-
-  final StringBuffer output = StringBuffer();
-  bool started = false; // Have we started writing a raw string.
-  for (int i = 0; i < s.length; i++) {
-    if (s[i] == "'") {
-      if (started)
-        output.write("'");
-      output.write(' "\'" ');
-      started = false;
-    } else if (!started) {
-      output.write("r'${s[i]}");
-      started = true;
-    } else {
-      output.write(s[i]);
-    }
-  }
-  if (started)
-    output.write("'");
-  return output.toString();
+///
+/// Strings with newlines are not supported.
+String generateString(String value) {
+  assert(!value.contains('\n'));
```

Comment:
Hi @a14n, could you explain why this newline check was added?

Reply:
I think I added this assert because the old method didn't work for `s` with a `\n` inside.

Reply:
The new method appears to handle any backslash character by generating a raw string. Maybe this assert isn't really needed.

Reply:
The assert is currently needed because the method does not handle `\n`. We could change it to use triple quote is `s` contains `\n` but it's a different handling than `\\`

Reply:
You're right; it's newline characters that are disallowed, not a backslash followed by an 'n'.

Values in JSON files - and ARB files are JSON files -  cannot contain newline characters, so it makes sense that this method  doesn't allow values to contain newline characters.